136 reset document form btn v2

stanislawK opened the pull request:
fixes #136 
Original discussion and screenshots in PR created by pushing directly to alinka repo https://github.com/CodeForPoznan/alinka-pyside/pull/139. That is the version pushed from the fork

stanislawK committed:
136 - Add btn to reset document form

stanislawK committed:
136 - Add btn to reset document form - unit tests

stanislawK committed:
136 - add content_container property to ApplicationFooterContainer

stanislawK committed:
136 - resolving conflicts

stanislawK committed:
136 - resolving conflicts

stanislawK committed:
136 - removing duplicated logic

Rafalkufel commented on alinka/widget/containers/main_body/content/application/application_tabs/application_tab/__init__.py:
```diff
@@ -62,6 +61,15 @@ def change_application_subject(self):
         self.application_reason_2.setFixedHeight(0)
 
         self.application_period.combobox.clear()
+        self.populate_application_reason(application_subject)
+
+    def populate_application_subject(self) -> None:
+        for issue, description in ISSUE_DESCRIPTION_NOMINATIVE_MAPPER.items():
+            self.application_subject.combobox.addItem(description, issue)
+
+    def populate_application_reason(self, application_subject: LabeledComboBoxComponent | None = None) -> None:
+        if not application_subject:
+            application_subject = self.application_subject.combobox.currentData()
```
:clap:

Rafalkufel commented on alinka/widget/containers/main_body/content/application/application_tabs/child_tab/school_data_group.py:
```diff
@@ -73,9 +73,12 @@ def refresh_school_dropdown(self):
+        self.school_type.combobox.addItems(SchoolTypes.values())
+        self.student_checkbox.checkbox.setChecked(False)
+        self.school_klass.clear()
+        self.school_profession.clear()
```
Why this change?

Rafalkufel commented on alinka/widget/containers/main_body/footer/application.py:
```diff
@@ -8,11 +8,16 @@ class ApplicationFooterContainer(QFrame):
-        layout.addWidget(self.print_btn)
+        self.clear_application_btn = QPushButton("Wyczyść formularz", self)
+        self.clear_application_btn.setStyleSheet("background-color: red; color: white;")
+        self.clear_application_btn.clicked.connect(self.content_container.application_container.clear)
```
WDYT to create dedicated hook in this class (in `self.clear_application`), and place all logic there

```
def clear_application(self):
    content_container = self.footer_container.main_body_container.content_container
    content_container.application_container.clear()
```
It will be more consistent with the rest of the code.

Rafalkufel commented on tests/test_clear_ApplicationContainer.py:
```diff
@@ -0,0 +1,352 @@
+import os
```
Very first test of our widget! :fireworks:

stanislawK committed:
136 - reset document - addressing CR

Rafalkufel commented on alinka/widget/containers/main_body/footer/application.py:
```diff
@@ -8,11 +8,16 @@ class ApplicationFooterContainer(QFrame):
     def __init__(self, parent: QWidget, visible: bool = False):
         super().__init__(parent)
         self.footer_container = parent
+        self.content_container = self.footer_container.main_body_container.content_container
```
Do  you really need to have `self.content_container`? Maybe you can move it to `clear_application`
```
    def clear_application(self):
       content_container = self.footer_container.main_body_container.content_container
       content_container.application_container.clear()

stanislawK commented:
Pcks are available for testing in https://github.com/stanislawK/alinka-pyside/actions/runs/18808039156

stanislawK committed:
136 - resolving conflicts

stanislawK committed:
clear_application_btn style changes

stanislawK commented:
To make that change in line with a new style, I removed red color and added pop-up window instead. I am not sure about the wording.
<img width="1791" height="1006" alt="Screenshot from 2025-12-04 00-09-59" src="https://github.com/user-attachments/assets/2f22c5ba-73eb-45e4-a2f0-cc37982afc06" />
